### DISCUSSION TITLE
Radio button won't show as checked

### DISCUSSION THREAD

**Discussion by mihaimazareanu** [neuefische/questions#342] [Web] [answered]
Hi,

I have implemented 2 radio buttons in my form, with one of them being checked by default.

    <label>
            <input
              type="radio"
              name="register"
              value=""
              checked="checked"
              onClick={handleClickParent}
            />
            Parent login
          </label>

    <label>
            <input type="radio" name="register" onClick={handleClickChild} />
            Child login
          </label>

And these are the 2 onClick functions for them:

    const handleClickParent = () => {
    setChild(false);
    setParent(true);
     };

    const handleClickChild = () => {
    setChild(true);
    setParent(false);
    };

They each use 2 state variables `parent` and `child` which are `false` in the initial state.

The functions work properly:

    {parent && (
        <>
          <DetailsFieldset>
            ...
          </DetailsFieldset>)}

    {child && (
        <ChildLoginSection>
          ...
        </ChildLoginSection>
      )}

So when I click the `child login` radio button the app renders the section it should when `child` is `true` and the same for `parent login`. However, the `child login` radio button won't get checked on the first click, only on the second click. This has to do obviously with the default `checked` property of `parent login` radio button, because if I deactivate it, the radio buttons work properly. So my understanding is that at the first click on `child login` Next.js (or better said the react part) will re-render everything, meaning the default `checked` property will be active first, although the onClick function works properly.

But I have no idea how to solve this issue, because I want `parent login` to be activated by default. Any ideas?

**Comment by actyralf** [comment, marked as answer]
The `checked` property of your radio buttons should react on your state, e.g. have a state value called `loginMode` with a default value of `"parent"`. Then your checked property could look like this: `checked={loginMode==="parent"?true:false}`

**Reply by mihaimazareanu** [reply]
Thank you Ralf, this solution helped me make my code even better, because before I had 2 state variables `parent` and `child` and now I can only use one, `loginMode`. 🙏
